zerver/tests/test_templates.py

```diff
@@ -0,0 +1,72 @@
+# -*- coding: utf-8 -*-
+from __future__ import absolute_import
+
+from django.test import TestCase
+from django.template.loader import get_template
+
+from zerver.models import get_user_profile_by_email
+from zerver.lib.test_helpers import get_all_templates
+
+
+class get_form_value(object):
+    def __init__(self, value):
+        self._value = value
+
+    def value(self):
+        return self._value
+
+
+class DummyForm(dict):
+    pass
+
+
+class TemplateTestCase(TestCase):
+    """
+    Tests for checking the rendering of backend templates.
+    """
+
+    def test_templates(self):
+        email = "hamlet@zulip.com"
+        user_profile = get_user_profile_by_email(email)
+
+        # `shallow_tested` is used to tell the receiver that the
+        # template was not rendered in an actual logical test.
+        # Note: `context` just holds dummy values to make the test
+        # pass. This test only ensures that the templates do not
+        # return a 500 error. It doesn't test the logic of the
+        # templates.
```

Comment:
Maybe worth explaining here that this test works by passing a set of arguments that cover the required context for most of Zulip's backend templates, and that this may need to be updated when we add new parameters to contexts used in templates